### PR TITLE
Feature/dependant move

### PR DESCRIPTION
**Pull request by marvin-bitterlich**
Please review #1065 and #1064 before this one!
Review please @Minecolonies/maintainers

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 207 issues
* ![MAJOR][MAJOR] 123 major
* ![MINOR][MINOR] 63 minor
* ![INFO][INFO] 21 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L472](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L472): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L627](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L627): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L728](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L728): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L886](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L886): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.224 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 83 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L181](https://github.com/Minecolonies/minecolonies/blob/64096e96c86b6ca43c3404c3f54af6df454b5868/src/main/java/com/minecolonies/coremod/colony/Colony.java#L181): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 197 issues
* ![MAJOR][MAJOR] 120 major
* ![MINOR][MINOR] 62 minor
* ![INFO][INFO] 15 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L472](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L472): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L627](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L627): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L728](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L728): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L886](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L886): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.224 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 83 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L181](https://github.com/Minecolonies/minecolonies/blob/2bfbe5ffda69a461f4be6b210fec48a1f28e0bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L181): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1066/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/api/java/com/minecolonies/api/util/InventoryUtils.java [hunk 1]
```diff
@@ -1,7 +1,6 @@
-package com.minecolonies.coremod.util;
+package com.minecolonies.api.util;
```
Comment (marchermans) [inline src/api/java/com/minecolonies/api/util/InventoryUtils.java:1]: Overruled. Utility Classes are supposed to be treated this way, and as such are God Classes by Definition.

File: src/api/java/com/minecolonies/api/util/Utils.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
-package com.minecolonies.coremod.util;
+package com.minecolonies.api.util;
```
Comment (marchermans) [inline src/api/java/com/minecolonies/api/util/Utils.java:1]: Overruled, Same reason as above.

File: src/main/java/com/minecolonies/coremod/util/WorkerUtil.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import net.minecraft.pathfinding.PathPoint;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3i;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Utility methods for BlockPos.
+ */
+public final class WorkerUtil
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/util/WorkerUtil.java:17]: Nice that this is finally split off from Entity utils.

However once enough abstraction has been done. This class should be moved into the API as it contains some usefull stuff (like Setting the SpawnPoint etc)

**Review by marchermans — APPROVED** [review]

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 197 issues
* ![MAJOR][MAJOR] 120 major
* ![MINOR][MINOR] 62 minor
* ![INFO][INFO] 15 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L472](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L472): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L627](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L627): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L728](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L728): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L886](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L886): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.224 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 83 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L181](https://github.com/Minecolonies/minecolonies/blob/651ccc6f0ebd54fa4e9ba25d3643596cec95e454/src/main/java/com/minecolonies/coremod/colony/Colony.java#L181): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 197 issues
* ![MAJOR][MAJOR] 120 major
* ![MINOR][MINOR] 62 minor
* ![INFO][INFO] 15 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L472](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L472): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L627](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L627): Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L728](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L728): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [WindowTownHall.java#L886](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java#L886): Reduce this anonymous class number of lines from 40 to at most 20, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [CitizenData.java#L23](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L23): class "CitizenData" has 43 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.224 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 83 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L181](https://github.com/Minecolonies/minecolonies/blob/1001b683940f44e48c938833c2f470d086a16575/src/main/java/com/minecolonies/coremod/colony/Colony.java#L181): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**